File: demo/src/main/java/com/example/demo/models/Message.java
package com.example.demo.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String content;

    private LocalDateTime timestamp;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "from_user_id")
    private Utilisateur fromUser;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "to_user_id")
    private Utilisateur toUser;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "parent_message_id")
    private Message parentMessage;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "conversation_id")
    private Conversation conversation;

    @PrePersist
    protected void onCreate() {
        this.timestamp = LocalDateTime.now();
    }
}

File: OccaPlace_Angular/src/app/core/model/Message.ts
// message.model.ts
import { Utilisateur } from './Utilisateur';

export interface Message {
  id: number;
  content: string;
  timestamp: Date; // Keep as Date for processing
  fromUser?: Utilisateur;
  toUser?: Utilisateur;
  parentMessage?: Message;
  status?: 'SENDING' | 'SENT' | 'FAILED';
}
